### PR TITLE
[Storage][Pruner][Fix] When catch up, the db version might be larger than the start version, do not check it.

### PR DESCRIPTION
**Pull request by grao1991**

### CODE REVIEW

File: storage/aptosdb/src/pruner/ledger_pruner/transaction_pruner.rs [hunk 1]
```diff
@@ -65,9 +65,25 @@ impl TransactionPruner {
...
-            .get_transaction_iter(start, (end - start) as usize)?
-            .collect()
+    ) -> Result<Vec<Transaction>> {
+        ensure!(end >= start);
+
+        let mut iter = self
+            .ledger_db
+            .transaction_db()
+            .iter::<TransactionSchema>(ReadOptions::default())?;
+        iter.seek(&start_version)?;
```
Comment (areshand) [inline storage/aptosdb/src/pruner/ledger_pruner/transaction_pruner.rs:75]: start instead of start_version?

Comment (grao1991) [reply storage/aptosdb/src/pruner/ledger_pruner/transaction_pruner.rs]: Done

**Review by areshand — APPROVED** [review]

File: storage/aptosdb/src/pruner/ledger_pruner/transaction_pruner.rs [hunk 2]
```diff
@@ -65,9 +68,24 @@ impl TransactionPruner {
...
-            .collect()
+    ) -> Result<Vec<Transaction>> {
+        ensure!(end >= start);
+
+        let mut iter = self
+            .transaction_db
+            .iter::<TransactionSchema>(ReadOptions::default())?;
+        iter.seek(&start)?;
+
+        // The capacity is capped by the pruning window.
```
Comment (msmouse) [inline storage/aptosdb/src/pruner/ledger_pruner/transaction_pruner.rs:79]: "pruning window" is confusing here.

**Review by msmouse — APPROVED** [review]

File: storage/aptosdb/src/pruner/ledger_pruner/transaction_pruner.rs (hunk 2 above)
Comment (grao1991) [reply storage/aptosdb/src/pruner/ledger_pruner/transaction_pruner.rs]: Done

**Pull request merged**